File: src/main/java/org/beangle/maven/container/Dependency.java
package org.beangle.maven.container;

public class Dependency {

  String groupId;

  String artifactId;

  public boolean matches(Dependency other) {
    return ((groupId.equals("*") || groupId.equals(other.groupId)) && (artifactId.equals("*") || artifactId
        .equals(other.artifactId)));
  }

  public Dependency(String groupId, String artifactId) {
    this.groupId = groupId;
    this.artifactId = artifactId;
  }
}

File: plugin/src/main/scala/org/beangle/maven/plugin/container/Dependency.scala
package org.beangle.maven.plugin.container

class Dependency(val groupId: String, val artifactId: String) {

  def matches(other: Dependency): Boolean = {
    ((groupId == "*" || groupId == other.groupId) && (artifactId == "*" || artifactId == other.artifactId))
  }
}
